packages/label-sync/src/label-sync.ts

```diff
@@ -94,15 +97,15 @@ export = (app: Application) => {
 };
 
 async function reconcileLabels(github: GitHubAPI, owner: string, repo: string) {
-  const newLabels = await getLabels();
+  const newLabels = await getLabels(github);
   const res = await github.issues.listLabelsForRepo({
     owner,
     repo,
     per_page: 100,
   });
   const oldLabels = res.data;
   const promises = new Array<Promise<unknown>>();
-  newLabels.labels.forEach(l => {
+  for (const l of newLabels.labels) {
```

Comment:
I want to poke at this just a little bit:

> Promise.all() created a race condition where deletion and creation could potentially happen in conjunction; I switched the logic to being synchronous for the time being (I think for how many labels we have, and how frequently this will run, this is a reasonable decision for the time being).

The labels that are being deleted are always different from the ones being created or update.  That is to say - the deletes should never interfere with the create/updates because they can never be on the same resources.  I've been running this script for over a year (albeit manually) with no issues.  Did you experience a problem here, or did you notice a potential for problems?

Reply:
I was seeing a lot of exceptions running the bot locally, I believe something long the lines of `VALIDATION_ERROR` (_actual logs aren't handy, but I can find them tomorrow).

My hunch was that:

* since there's no guarantee of the ordering of `Promise.all`, perhaps requests were being queued that were somehow contradictory with each other?
* or, perhaps this is more likely, the errors were cropping up when too many operations where happening concurrently?

In general, I'm not a huge proponent of an unbounded `Promise.all`, just because there's nothing that limits the amount of concurrency. An alternative refactor might be to figure out a way to kick off and await `N` promises at  time.

```js
let count  = 0
const work = []
for (const l of labels) {
  work.push(l)
  if ((i % 4) === 0) await Promise.all(work)
  count++
}
await Promise.all(work)
```

I can test more tomorrow with whatever approach you land on.